astroquery/mast/cutouts.py

```diff
@@ -637,3 +637,163 @@ def get_cutouts(self, coordinates, *, size=5, survey=None):
 
 
 Zcut = ZcutClass()
+
+
+class HapcutClass(MastQueryWithLogin):
+    """
+    MAST Hubble Advanced Product (HAP) cutout query class.
+    
+    Class for accessing HAP image cutouts.
+    """
+
+    def __init__(self):
+
+        super().__init__()
+
+        services = {"astrocut": {"path": "astrocut"}}
+
+        self._service_api_connection.set_service_params(services, "hapcut")
+
+    def download_cutouts(self, coordinates, *, size=5, path=".", inflate=True):
```

Comment:
There seem to be a lot of duplicates in this method shared amongst the different cutout classes, I wonder maybe it's time to consolidate them into a base class, so any future maintenance becomes easier